pkg/snapshotmgr/snapshot_manager.go

```diff
@@ -244,65 +253,97 @@ func (this *SnapshotManager) createSnapshot(peID astrolabe.ProtectedEntityID, ta
 		return updatedPeID, nil
 	}
 
+	_, err = this.UploadSnapshot(peID, updatedPeID, peSnapID, pe, ctx, backupRepositoryName, snapshotRef)
+	if err != nil {
+		return astrolabe.ProtectedEntityID{}, err
+	}
+
+	return updatedPeID, nil
+}
+
+/*
+Creates an Upload CR
+*/
+func (this *SnapshotManager) UploadSnapshot(peID astrolabe.ProtectedEntityID, updatedPeID astrolabe.ProtectedEntityID,
+	peSnapID astrolabe.ProtectedEntitySnapshotID, pe astrolabe.ProtectedEntity, ctx context.Context, backupRepositoryName string,
+	snapshotRef string) (*v1api.Upload, error) {
 	this.Info("Start creating Upload CR")
 	config, err := rest.InClusterConfig()
 	if err != nil {
 		this.WithError(err).Errorf("Failed to get k8s inClusterConfig")
-		return updatedPeID, err
+		return nil, err
 	}
 	pluginClient, err := plugin_clientset.NewForConfig(config)
 	if err != nil {
 		this.WithError(err).Errorf("Failed to get k8s clientset from the given config: %v ", config)
-		return updatedPeID, err
+		return nil, err
 	}
 
 	// look up velero namespace from the env variable in container
 	veleroNs, exist := os.LookupEnv("VELERO_NAMESPACE")
 	if !exist {
 		this.WithError(err).Errorf("CreateSnapshot: Failed to lookup the env variable for velero namespace")
-		return updatedPeID, err
+		return nil, err
 	}
 
-	uploadBuilder := builder.ForUpload(veleroNs, "upload-"+peSnapID.GetID()).
+	uploadUUID, err := uuid.NewRandom()
+	if err != nil {
+		this.WithError(err).Errorf("Could not create upload UUID")
+		return nil, err
+	}
+	uploadName := "upload-" + uploadUUID.String()
+	this.Info("Creating Upload CR: %s/%s", veleroNs, uploadName)
+	uploadBuilder := builder.ForUpload(veleroNs, uploadName).
 		BackupTimestamp(time.Now()).
 		NextRetryTimestamp(time.Now()).
-		Phase(v1api.UploadPhaseNew).
-		SnapshotReference(snapshotRef)
```

Comment:
This line `SnapshotReference(snapshotRef)` should be added back.  I didn't rebase when I submitted the PR so I didn't pick up this one from the master.

Reply:
Funny, I thought I had.  Maybe there were two.

Reply:
OK, I had modified the method to take in the snapshotRef but then lost the place where it actually used it.  Fixed, thanks!